src/store.js

```diff
@@ -58,6 +58,20 @@ export function expKeyStatesHas(keyStates, stateName, key, prefix) {
   return hasIt;
 }
 
+export function setConfigLoadedKeys(keyStates, exp) {
+  const clean = objects.assign({}, exp);
+  delete clean.id;
+  const expLoaded = new Set();
+  const expMap = new Map();
+  expMap.set('loaded', expLoaded)
+  keyStates.experiments.set(exp.id, expMap);
+  objects.flattenKeys(clean, function(key) {
+    return!strings.startsWith(key, '_') || key === '_values' || key === '_initializers'
+  }).filter(function(key) { return strings.endsWith(key, '_values') || strings.endsWith(key, '_initializers') })
+  .forEach(function(key) { expLoaded.add(key.replace(/._values|._initializers/gi, '')) });
+}
+
+
```

Comment:
Nitpick: less dense
```suggestion
export function setConfigLoadedKeys(keyStates, exp) {
  const clean = objects.assign({}, exp);
  delete clean.id;

  const expLoaded = new Set();
  const expMap = new Map();

  expMap.set('loaded', expLoaded)
  keyStates.experiments.set(exp.id, expMap);

  objects.flattenKeys(clean, function(key) {
    return !strings.startsWith(key, '_') || key === '_values' || key === '_initializers'
  })
    .filter(function(key) {
      return strings.endsWith(key, '_values') || strings.endsWith(key, '_initializers')
    })
    .forEach(function(key) {
      expLoaded.add(key.replace(/._values|._initializers/gi, ''))
    });
}


```

Reply:
Also, no need to change. However, a pattern that is used often in JavaScript to avoid assignment with loops is `reduce()`. It often cuts out several lines and avoids side effects.

```
	const expLoaded = objects
		.flattenKeys(clean, function(key) {
			return !strings.startsWith(key, '_') || key === '_values' || key === '_initializers'
		})
		.reduce(function(acc, key) {
			return acc.add(key.replace(/._values$|._initializers$/gi, ''));
		}, new Set());
```